youtube_dl/extractor/hketv.py

```diff
@@ -0,0 +1,157 @@
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        title = self._html_search_meta('ed_title', webpage, fatal=True)
+
+        file_id = self._html_search_regex(r'post_var\["file_id"\]\s*=\s*(.+?);', webpage, 'file ID')
+        curr_url = self._html_search_regex(r'post_var\["curr_url"\]\s*=\s*"(.+?)";', webpage, 'curr URL')
+        data = {
+            'action': 'get_info',
+            'curr_url': curr_url,
+            'file_id': file_id,
+            'video_url': file_id,
+        }
+        _APPS_BASE_URL = 'https://apps.hkedcity.net'
+        handler_url = _APPS_BASE_URL + '/media/play/handler.php'
+
+        response = self._download_json(
+            handler_url, video_id,
+            data=urlencode_postdata(data),
+            headers=merge_dicts({'Content-Type': 'application/x-www-form-urlencoded'},
+                                self.geo_verification_headers()))
+
+        result = response.get('result')
+
+        formats = []
+        subtitles = {}
+
+        if response.get('success') and response.get('access'):
+            width = str_to_int(result.get('width'))
+            height = str_to_int(result.get('height'))
+
+            playlist0 = try_get(result, lambda x: x['playlist'][0], dict)
+            fmts = try_get(playlist0, lambda x: x['sources'], list)
+            for fmt in fmts:
+                label = fmt.get('label')
+                if label == 'HD':
+                    h = 720
+                elif label == 'SD':
+                    h = 360
+                w = h * width // height
+                urlh = self._downloader.urlopen(_APPS_BASE_URL + fmt.get('file'))
+                formats.append({
+                    'format_id': label,
+                    'ext': fmt.get('type'),
+                    'url': urlh.geturl(),
+                    'width': w,
+                    'height': h,
+                })
+
+            tracks = try_get(playlist0, lambda x: x['tracks'], list)
+            for track in tracks:
+                if not isinstance(track, dict):
+                    continue
+                track_kind = track.get('kind')
+                if not track_kind or not isinstance(track_kind, compat_str):
+                    continue
+                if track_kind.lower() not in ('captions', 'subtitles'):
+                    continue
+                track_url = _APPS_BASE_URL + track.get('file')
+                if not track_url:
+                    continue
+                track_label = track.get('label')
+                subtitles.setdefault(self._CC_LANGS.get(track_label, track_label), []).append({
+                    'url': self._proto_relative_url(track_url),
+                    'ext': 'srt',
+                })
+
+        else:
+            error = clean_html(response.get('access_err_msg'))
+            if 'Video streaming is not available in your country' in error:
+                raise GeoRestrictedError(error)
```

Comment:
`raise_geo_restricted` and specify `countries`.

Reply:
Thanks!  After consulting other extractors, I am now using this:
```python
                self.raise_geo_restricted(msg=error, countries=self._GEO_COUNTRIES)
```

with the following definitions near the top:
```python
    _GEO_BYPASS = False
    _GEO_COUNTRIES = ['HK']
```